sklearn/neighbors/_base.py

```diff
@@ -34,6 +34,7 @@
 from ..utils.multiclass import check_classification_targets
 from ..utils.validation import check_is_fitted
 from ..utils.validation import check_non_negative
+from ..utils._param_validation import Interval, Integral, Real, StrOptions
```

Comment:
I think it would be better if `Integral` and `Real` are imported from `numbers` where they are defined.

Reply:
Fixed the import directly from `numbers`